android/expoview/src/main/java/host/exp/exponent/ExpoUpdatesAppLoader.java

```diff
@@ -107,16 +125,20 @@ public void start() {
 
     File directory;
     try {
-      directory = UpdatesUtils.getOrCreateUpdatesDirectory(mContext);
+      directory = UpdatesUtils.getOrCreateUpdatesDirectory(context);
     } catch (Exception e) {
       onError(e);
       return;
     }
 
-    startLoaderTask(configuration, directory, selectionPolicy);
+    startLoaderTask(configuration, directory, selectionPolicy, context);
   }
 
-  private void startLoaderTask(final UpdatesConfiguration configuration, final File directory, final SelectionPolicy selectionPolicy) {
+  private void startLoaderTask(final UpdatesConfiguration configuration, final File directory, final SelectionPolicy selectionPolicy, final Context context) {
+    mUpdatesConfiguration = configuration;
+    mUpdatesDirectory = directory;
+    mSelectionPolicy = selectionPolicy;
```

Comment:
Can an AppLoader be started twice? I'm mostly concerned about the values of these member variables changing over time.

Reply:
In a couple situations the AppLoader might restart itself with slightly different configuration values (e.g. if the experience is recovering from an error it will force an update check), but in practice this only happens before the app is actually launched and these values are queried externally.

Reply:
We could potentially have the AppLoader create and start a new instance instead, in that case, and re-register itself in the kernel's `mManifestUrlToAppLoader` map, and then explicitly protect against an AppLoader starting more than once.

Reply:
implemented in a4fdf4e